concordium-consensus/tests/globalstate/GlobalStateTests/Account.hs

```diff
@@ -0,0 +1,143 @@
+                    tst' <- loadDirect ref
+                    inMemoryTST' <- persistentToInMemoryTST tst'
+                    liftIO $ assertEqual "stored then loaded token account state table should be equal" inMemoryTST' inMemoryTST
+
+-- | Verify hash consistency between in-memory and persisted TokenStateTable.
+testInMemoryEqualsPersistentHash :: Word -> SpecWith (PersistentBlockStateContext PV)
+testInMemoryEqualsPersistentHash lvl =
+    it "Hash of in-memory token account state table equals hash of persisted token account state table" $ \blobstore ->
+        withMaxSuccess (1000 * fromIntegral lvl) $
+            property $ do
+                inMemoryTST <- arbitraryInMemoryTokenStateTable
+                return $ ioProperty $ flip runBlobStoreT blobstore $ do
+                    let h :: TokenStateTableHash = getHash inMemoryTST
+                    tst <- inMemoryToPersistentTST inMemoryTST
+                    h' :: TokenStateTableHash <- getHashM tst
+                    liftIO $ assertEqual "Hash of in-memory/persisted account state table should be equal" h h'
+
+-- | Verify consistency between serialization/deserialization of TokenAccountState.
+testSerializeTokenAccountState :: Word -> Spec
+testSerializeTokenAccountState lvl =
+    it "Serialize, then deserialize of TokenAccountState results in identity" $
+        withMaxSuccess (10000 * fromIntegral lvl) $
+            forAll arbitraryTokenAccountState $ \state -> property $ do
+                let stateOrErr = runGet get $ runPut $ put state
+                case stateOrErr of
+                    Left err -> assertFailure $ "Deserialization failed: " ++ show err
+                    Right state' ->
+                        assertEqual "serialize . deserialize for token account state should be the identity" state state'
+
+tests :: Word -> Spec
+tests lvl = do
+    describe "GlobalStateTests.Account" $
+        do
+            testSerializeTokenAccountState lvl
+            -- tests that need access to a temporary blockstate
+            around
+                ( \kont ->
+                    withTempDirectory "." "blockstate" $ \dir ->
+                        bracket
+                            ( do
+                                pbscBlobStore <- createBlobStore (dir </> "blockstate.dat")
+                                pbscAccountCache <- PA.newAccountCache 100
+                                pbscModuleCache <- M.newModuleCache 100
+                                pbscAccountMap <- LMDBAccountMap.openDatabase (dir </> "accountmap")
+                                return PersistentBlockStateContext{..}
+                            )
+                            ( \PersistentBlockStateContext{..} -> do
+                                closeBlobStore pbscBlobStore
+                                LMDBAccountMap.closeDatabase pbscAccountMap
+                            )
+                            kont
+                )
+                $ do
```

Comment:
I suggest using the in-memory BlobStore implementation for tests, as it will be faster. See `GlobalStateTests.AccountList` for example.